### PR TITLE
feat: Support rowSelection.dirty

### LINKED ISSUE #24640: [Table AJAX] Row selection has been reset when changing pagination
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[Edit on CodeSandbox](https://codesandbox.io/s/bug-table-ajax-selection-pagination-ant-design-demo-gw7cl?file=/index.js)

### Steps to reproduce
1. Select a row
2. Next page
3. Select  another a row
4. Check console.log

### What is expected?
The row selected keys should keep when changing the page

### What is actually happening?
The row selected keys have been reset

| Environment | Info |
|---|---|
| antd | 4.3.0 |
| React | 16.13.1 |
| System | OSX |
| Browser | Chrome |

### PR DESCRIPTION
**Pull request by zombieJ**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Perfermance optimization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

resolve #24640
resolve #24718

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |    Table support `rowSelection.dirty` to enable cache `key` with ajax.       |
| 🇨🇳 Chinese |     Table 添加 `rowSelection.dirty` 以支持异步数据下保留 `key`。     |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

-----
[View rendered components/table/demo/ajax.md](https://github.com/ant-design/ant-design/blob/table-dirty/components/table/demo/ajax.md)
[View rendered components/table/index.en-US.md](https://github.com/ant-design/ant-design/blob/table-dirty/components/table/index.en-US.md)
[View rendered components/table/index.zh-CN.md](https://github.com/ant-design/ant-design/blob/table-dirty/components/table/index.zh-CN.md)

### CODE REVIEW

File: components/table/demo/ajax.md [hunk 1]
```diff
@@ -11,12 +11,16 @@ title:
...
 
+当使用 `rowSelection` 时，请设置 `rowSelection.dirty` 属性以保留 `key`。
+
 **注意，此示例使用 [模拟接口](https://randomuser.me)，展示数据可能不准确，请打开网络面板查看请求。**
 
 ## en-US
 
 This example shows how to fetch and present data from a remote server, and how to implement filtering and sorting in server side by sending related parameters to server.
 
+Setting `rowSelection.dirty` to keep the `key` when enable selection.
```
Comment (afc163) [inline components/table/demo/ajax.md:22]: `reserveSelectedKeys={true}` or `clearSelectedKeysOnDataSourceChanged={false}`?

Comment (zombieJ) [reply components/table/demo/ajax.md:22]: `rowSelection.reserveKeys` 好了，`rowSelection` 本身是个 object 不单独放外面了。

Comment (afc163) [reply components/table/demo/ajax.md:22]: rowSelection.reserveSelectedKeys

**Comment by zombieJ** [comment]
updated

File: components/table/interface.tsx [hunk 2]
```diff
@@ -125,6 +125,8 @@ export type SelectionSelectFn<T> = (
 ) => void;
 
 export interface TableRowSelection<T> {
+  /** Keep the selection keys in list even the key not exist in `dataSource` anymore */
+  preserveKeys?: boolean;
```
Comment (afc163) [inline components/table/interface.tsx:129]: preserveSelectedRowKeys 吧。

Comment (zombieJ) [reply components/table/interface.tsx]: 👌

**Comment by zombieJ** [comment]
done

**Review by afc163 — APPROVED** [review]

**Pull request merged**